### PR TITLE
[MARKET-320] Return cache control headers for /plugins

### PR DESCRIPTION
**Pull request by nantunes**
IE10 was being more aggressive with cache, so every refresh resulted in 304 / NOT MODIFIED.
The user received no feedback, altrough the request was made.

Added header `Cache-Control: no-cache, no-store` to the response.

@graimundo

### CODE REVIEW

File: marketplace-core/src/main/java/org/pentaho/marketplace/endpoints/MarketplaceService.java [hunk 1]
```diff
@@ -75,7 +76,7 @@ public StringOperationResultDTO hello() {
   @GET
   @Path( "/plugins" )
   @Produces( MediaType.APPLICATION_JSON )
-  public IterablePluginOperationResultDTO getPlugins() {
+  public Response getPlugins() {
```
Comment (graimundo) [inline marketplace-core/src/main/java/org/pentaho/marketplace/endpoints/MarketplaceService.java]: Do not change the method return type. If you need to change the HTTP properties of the Response add "@Context HttpServletResponse response" to the method arguments. See DET for an example: https://github.com/webdetails/det/blob/master/pentaho-det-core/src/main/java/org/pentaho/det/impl/endpoints/DataSourceService.java#L72-L83

Comment (nantunes) [inline marketplace-core/src/main/java/org/pentaho/marketplace/endpoints/MarketplaceService.java]: Ok. I initially did that, but the project don't have javax.servlet dependency. So I tryed to keep in the scope of JAX-RS. I'll add the dependency, then.

Comment (graimundo) [inline marketplace-core/src/main/java/org/pentaho/marketplace/endpoints/MarketplaceService.java]: I see your point and both approaches have pros and cons. For the time being add the javax.servlet dependency in order to keep the return type explicit.

Comment (nantunes) [inline marketplace-core/src/main/java/org/pentaho/marketplace/endpoints/MarketplaceService.java]: Ok @graimundo. Please review.

**Comment by graimundo** [comment]
@nantunes please update for servlet 3 and test with latest builds

**Comment by nantunes** [comment]
@graimundo I tested it and committed the new artifactId.

**Pull request merged**